src/main/java/mate.academy/service/impl/ProductServiceImpl.java

```diff
@@ -2,15 +2,26 @@
 
 import java.util.List;
 import java.util.stream.Collectors;
+import mate.academy.lib.Component;
+import mate.academy.lib.Inject;
 import mate.academy.model.Product;
 import mate.academy.service.FileReaderService;
 import mate.academy.service.ProductParser;
 import mate.academy.service.ProductService;
 
+@Component
 public class ProductServiceImpl implements ProductService {
+    @Inject
     private ProductParser productParser;
+
+    @Inject
     private FileReaderService fileReaderService;
 
+    public ProductServiceImpl() {
+        this.productParser = new ProductParserImpl();
+        this.fileReaderService = new FileReaderServiceImpl();
+    }
```

Comment:
This is not Injection))
```suggestion
```

Reply:
not fixed: ProductParser and FileReaderService should be injected by Injector.java
not by constructor. 

So, you don't implement the injector in the right way.